storageClient/storageClient.go

```diff
@@ -291,6 +291,24 @@ func (c *StorageClient) ShareRecordWithGroup(ctx context.Context, params ShareGr
 	return result, err
 }
 
+// GetSharedWithGroup returns a list of all records shared with a group
+func (c *StorageClient) GetSharedWithGroup(ctx context.Context, params ListGroupRecordsRequest) (*ExecutorQueryResponse, error) {
+	var result *ExecutorQueryResponse
+	path := c.Host + storageServiceBasePath + "/groups/" + params.GroupID.String() + "/share"
+	req, err := e3dbClients.CreateRequest("GET", path, params)
+	if err != nil {
+		return result, err
+	}
+	urlParams := req.URL.Query()
+	urlParams.Set("nextToken", strconv.Itoa(int(params.NextToken)))
+	if params.Max != 0 {
+		urlParams.Set("max", strconv.Itoa(int(params.Max)))
+	}
+	req.URL.RawQuery = urlParams.Encode()
```

Comment:
We also need support for query param that is a list of writer_ids to filter this list call with